src/Liquidator.sol

```diff
@@ -69,176 +69,172 @@ contract Liquidator is Ownable {
 
-        if (!forSale) {
-            return (0, address(0), false);
+        if (!inAuction) {
+            return (0, false);
         }
 
-        uint256 startPrice = (auctionInformation[vaultAddress].openDebt * 150) / 100;
-        uint256 surplusPrice = (auctionInformation[vaultAddress].openDebt * (150 - 100)) / 100;
-        uint256 priceDecrease = (surplusPrice * (block.number - auctionInformation[vaultAddress].startBlock))
-            / (hourlyBlocks * breakevenTime);
+        price = _calcPriceOfVault(auctionInformation[vault].startTime, auctionInformation[vault].openDebt);
+    }
 
-        totalPrice;
-        if (priceDecrease > startPrice) {
+    /**
+     * @notice Function returns the current auction price given time passed and the openDebt.
+     * @param startTime The timestamp when the auction started.
+     * @param openDebt The open debt taken by `originalOwner`.
+     * @return price The total price for which the vault can be purchased.
+     * @dev We use a dutch auction: price constantly decreases and the first bidder buys the vault
+     * And immediately ends the auction.
+     */
+    function _calcPriceOfVault(uint256 startTime, uint256 openDebt) internal view returns (uint256 price) {
+        uint256 auctionTime = block.timestamp - startTime; //Can be unchecked
+
+        if (auctionTime > maxAuctionTime) {
             //ヽ༼ຈʖ̯ຈ༽ﾉ
-            totalPrice = 0;
+            price = 0;
         } else {
-            totalPrice = startPrice - priceDecrease;
+            price = uint256(openDebt) * startPriceMultiplier * (maxAuctionTime - auctionTime) / maxAuctionTime / 100;
         }
-
-        return (totalPrice, auctionInformation[vaultAddress].baseCurrency, forSale);
     }
 
     /**
-     * @notice Function a user calls to buy the vault during the auction process. This ends the auction process
-     * @dev Ensure the vault is for sale before calling this function.
-     * @param vaultAddress the vaultAddress of the vault the user want to buy.
+     * @notice Function a user (the bidder) calls to buy the vault and end the auction.
+     * @param vault The contract address of the vault.
+     * @dev We use a dutch auction: price constantly decreases and the first bidder buys the vault
+     * And immediately ends the auction.
      */
-    function buyVault(address vaultAddress) public {
-        (uint256 priceOfVault,, bool forSale) = getPriceOfVault(vaultAddress);
-
-        require(forSale, "LQ_BV: Not for sale");
+    function buyVault(address vault) external {
+        AuctionInformation memory auctionInformation_ = auctionInformation[vault];
+        require(auctionInformation_.inAuction, "LQ_BV: Not for sale");
 
-        address trustedCreditor = auctionInformation[vaultAddress].trustedCreditor;
-        address asset = auctionInformation[vaultAddress].baseCurrency;
+        uint256 priceOfVault = _calcPriceOfVault(auctionInformation_.startTime, auctionInformation_.openDebt);
```

Comment:
```suggestion
        uint256 priceOfVault = _calcPriceOfVault(block.timestamp - auctionInformation_.startTime, auctionInformation_.openDebt);
```

Reply:
Don't think this is correct, see line 151